Sources/EdgeAgentUpdater/EdgeAgentUpdateService.swift

```diff
@@ -0,0 +1,77 @@
+import EdgeAgentGRPC
+import Foundation
+import Logging
+import NIOCore
+import NIOFoundationCompat
+import _NIOFileSystem
+
+struct EdgeAgentService: Edge_Agent_Updater_Services_V1_EdgeAgentUpdateService.ServiceProtocol {
+    let logger = Logger(label: "EdgeAgentUpdateService")
+    let binaryPath: FilePath
+
+    func updateAgent(
```

Comment:
this is writing the file. what about shutting down the edge-agent during update? or what happens if the new file doesn't start?

Reply:
Good call, I forgot about that. Let's discuss that a bit later today, I think we should make edge-agent a child process of edge-agent-updater. That way it controls the lifecycle